rfc-119-merge-mainstream-browse-topics-and-taxons.md

```diff
@@ -0,0 +1,169 @@
+- `top_level_browse_pages` (an be derived from `base_path` and following `browse_parent`)
+- `active_top_level_browse_page` (can be derived from `base_path`)
+- `second_level_browse_pages` (an be derived from `base_path` and following `browse_parent`)
+- `related_topics` (is the related specialist topic, and would become a self reference)
+- `primary_publishing_organisation` (redundant)
+- `available_translations` (redundant)
+
+Dropped from specialist topic:
+- `taxons` (is the taxon, would become a self reference)
+- `topic_taxonomy_taxons` (redundant)
+- `primary_publishing_organisation` (redundant)
+- `available_translations` (redundant)
+
+#### Details attributes
+
+Taxons continue to have the following details:
+- `internal_name`
+- `notes_for_editors`
+- `visible_to_departmental_editors`
+
+Added from mainstream browse:
+- `browse_groups` (the curated groupings of browse children)
+
+Added from specialist topic:
+- `specialist_topic_groups` (the curated groupings of specialist_topic_children)
+
+Dropped from mainstream browse:
+- `internal_name` (redundant)
+- `second_level_ordering` (can be deduced by following link to parent)
+- `ordered_second_level_browse_pages` (can be deduced by following link to parent)
+
+Dropped from specialist topics:
+- `internal_name` (redundant)
+
+### Example of proposed taxon content item after merging:
+```json
+{
+  "title": "Universal Credit",
+  "description": "Applying, signing into your account, and help with housing, disability, health conditions and unemployment",
+  "base_path": "/welfare/universal-credit",
+  "content_id": "62fcbba5-3a75-4d15-85a6-d8a80b03d57c",
+  "document_type": "taxon",
+  "schema_name": "taxon",
+  "publishing_app": "content-tagger",
+  "rendering_app": "collections",
+  "locale": "en",
+  "phase": "live",
+  "first_published_at": "2018-03-08T16:38:00.000+00:00",
+  "public_updated_at": "2018-08-22T13:14:40.000+00:00",
+  "updated_at": "2019-08-09T20:07:42.761Z",
+  "publishing_request_id": "12821-1535466331.861-10.3.3.1-394",
+  "publishing_scheduled_at": null,
+  "scheduled_publishing_delay_seconds": null,
+  "analytics_identifier": null,
+  "withdrawn_notice": {},
+  "links": {
+    "parent_taxons": [],
+    "browse_children": [],
+    "browse_parent": [],
+    "specialist_topic_children": [],
+    "specialist_topic_parent": [],
+    "primary_publishing_organisation": [],
+    "available_translations": []
+  },
+  "details": {
+    "internal_name": "Universal Credit [M]",
+    "notes_for_editors": "",
+    "visible_to_departmental_editors": false,
+    "specialist_topic_groups": [],
+    "browse_groups": [],
+    "browse_ordering": "curated",
+    "ordered_second_level_browse_pages": []
+  }
+}
+```
+
+We would probably approach this work in two stages by merging mainstream browse *and then* specialist topics. After both are merged we'd look at deprecating mainstream browse and specialist topics content items in the Publishing API and Content Store.
```

Comment:
I've been wondering about this in relation to [the comments above about self-referential links](https://github.com/alphagov/govuk-rfcs/pull/119/files#diff-b26d5a4ca9078275cfb5727792c332e19f06339d723e85ccec934c15d9e9668eR71) - so seems like if we stage this we might want to modify some of the removals or create some initial duplication that we only clear up in a final pass?

Is that still easier than trying to munge these both together at the same time?

I can imagine both at once creates quite a lot of frontend and publishing work along-side the data reconciliation?

Reply:
@huwd have talked outside these comments. Think we agreed it may be helpful to have an examples of what the taxon content item looks after mainstream browse is added, but before specialist topic is. Is this something you could add @WorkSutton ?